[caffe2][be] migrate gloabl static initializer

klein-meta opened the pull request:
Summary:
Caffe2 lib has 200+ global static initializer usage, which are papar-cut reference to startup perf. Detail in this post https://fb.workplace.com/groups/arglassesperf/permalink/623909116287154.

This Diff migrate StorageImpl.cpp

Addtional Context: https://fb.workplace.com/groups/arglassesperf/permalink/623909116287154

Test Plan: CI

Differential Revision: D58639283

aaronenyeshi reviewed (approved):
Please fix lint before landing:
```
2024-06-19T15:56:10.5610812Z   Warning (CLANGFORMAT) format
2024-06-19T15:56:10.5611379Z     See https://clang.llvm.org/docs/ClangFormat.html.
2024-06-19T15:56:10.5611957Z     Run `lintrunner -a` to apply this patch.
2024-06-19T15:56:10.5612304Z 
2024-06-19T15:56:10.5612579Z     You can run `lintrunner -a` to apply this patch.
2024-06-19T15:56:10.5612975Z 
2024-06-19T15:56:10.5613465Z     10  10 | // A allowlist of device type, currently available is PrivateUse1.
2024-06-19T15:56:10.5614575Z     11  11 | ska::flat_hash_set<c10::DeviceType>& DeviceTypeAllowList() {
2024-06-19T15:56:10.5615633Z     12  12 |   static ska::flat_hash_set<c10::DeviceType> deviceTypeAllowList{
2024-06-19T15:56:10.5616623Z     12     |-    DeviceType::PrivateUse1};
2024-06-19T15:56:10.5617514Z         13 |+      DeviceType::PrivateUse1};
2024-06-19T15:56:10.5618278Z     14  14 |   return deviceTypeAllowList;
2024-06-19T15:56:10.5618859Z     15  15 | }
2024-06-19T15:56:10.5619319Z     16  16 | 
2024-06-19T15:56:10.5619703Z 
```

klein-meta committed:
[caffe2][be] migrate gloabl static initializer (#128784)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/128784

Caffe2 lib has 200+ global static initializer usage, which are papar-cut reference to startup perf. Detail in this post https://fb.workplace.com/groups/arglassesperf/permalink/623909116287154.

This Diff migrate StorageImpl.cpp

Addtional Context: https://fb.workplace.com/groups/arglassesperf/permalink/623909116287154

Test Plan: CI

Reviewed By: aaronenyeshi

Differential Revision: D58639283
